Sources/Codextended/Codextended.swift

```diff
@@ -129,6 +129,22 @@ public extension Decoder {
     }
 }
 
+// MARK: - Decoding Array
+
+public extension Decoder {
+    /// Decode an array ignoring elements that fail to decode.
+    func decodeArraySafely<T: Decodable>(_ key: String, as type: T.Type = T.self) throws -> [T] {
+        return try decodeArraySafely(AnyCodingKey(key), as: type)
+    }
+    
+    /// Decode an array ignoring elements that fail to decode.
+    func decodeArraySafely<T: Decodable, K: CodingKey>(_ key: K, as type: T.Type = T.self) throws -> [T] {
+        let container = try self.container(keyedBy: K.self)
+        let decodedArray = try container.decode([IgnoreFailureDecodable<T>].self, forKey: key)
+        return decodedArray.compactMap({ $0.element })
```

Comment:
Decoding with the custom `IgnoreFailureDecodable` and then `compactMap` is the only reliable way I know to do this. 
I've tried in the past to manually use the unkeyedContainer and loop to it, but if decoding fails the index doesn't advance and there is no way to advance it manually 😢